src/libs/conduit/conduit_data_array.cpp

```diff
@@ -229,93 +214,92 @@ DataArray<T>::diff(const DataArray<T> &array, Node &info, const float64 epsilon)
 
             if(res)
             {
-                log::error(info_meta, protocol, "data item(s) mismatch; see diff below");
+                log::error(info, protocol, "data item(s) mismatch; see 'value' section");
             }
         }
     }
 
-    log::validation(info_meta, !res);
+    log::validation(info, !res);
 
     return res;
 }
 
 //---------------------------------------------------------------------------//
 template <typename T> 
 bool
-DataArray<T>::equals(const DataArray<T> &array, Node &info, const float64 epsilon) const 
+DataArray<T>::diff_compatible(const DataArray<T> &array, Node &info, const float64 epsilon) const 
 { 
-    const std::string protocol = "dataarray::equals";
-    bool res = true;
+    const std::string protocol = "dataarray::diff_compatible";
+    bool res = false;
     info.reset();
 
-    // NOTE: The 'info' node is split into two sections to provide a diff digest
-    // and full diff details in separate parts of the node.
-    Node &info_meta = info.append();
-    Node &info_content = info.append();
-
     index_t t_nelems = number_of_elements();
     index_t o_nelems = array.number_of_elements();
 
-    if(t_nelems != o_nelems)
+    if(t_nelems > o_nelems)
     {
         std::ostringstream oss;
-        oss << "data length mismatch (" << t_nelems << "/" << o_nelems << ")";
-        log::error(info_meta, protocol, oss.str());
-        res = false;
+        oss << "arg data length incompatible (" << t_nelems << "/" << o_nelems << ")";
+        log::error(info, protocol, oss.str());
+        res = true;
     }
     else
     {
-        std::ostringstream oss;
-        oss << "data length match (" << t_nelems << ")";
-        log::info(info_meta, protocol, oss.str());
-
         if(dtype().is_char8_str())
         {
+            // TODO(JRC): Currently, due to the way that strings are represented
+            // in C/C++ (i.e. null-terminated), this comparison isn't very useful.
+            // Consider comparing "a" with "aa"; the first string has two elements
+            // (i.e. "a" and "\0"), so will come back true for this diff when
+            // compared against "aa" (whose first two elements are "a" and "a").
+            // This behavior should be updated if a reasonable and useful
+            // interpretation is discovered for string diffs.
             uint8 *t_compact_data = new uint8[dtype().bytes_compact()];
```

Comment:
i agree -- i don't think folks would expect this result, can we use `as_string()` to obtain the values and compare?
If it's not compact, we could call `compact_to()` and then us `as_string()`